src/core/DY.Core.Trace.Manipulation.fst

```diff
@@ -486,6 +534,56 @@ let rec get_state_aux_state_invariant #invs prin sess_id tr =
     | None -> ()
     | Some content -> state_pred.pred_later tr_init tr prin sess_id content
 
+
+val get_state_same_trace:
+  prin:principal -> sess_id:state_id -> tr:trace ->
+  Lemma
+  (ensures (
+    let (opt_content, tr_out) = get_state prin sess_id tr in
+    tr == tr_out
+    ))
+  [SMTPat (get_state prin sess_id tr);]
+let get_state_same_trace prin sess_id tr =
+  reveal_opaque (`%get_state) get_state
+
+
+val get_state_aux_state_was_set:
+  prin:principal -> sess_id:state_id -> tr:trace ->
+  Lemma
+  (ensures (
+    match get_state_aux prin sess_id tr with
+    | None -> True
+    | Some content -> 
+          state_was_set tr prin sess_id content
```

Comment:
Nit (indentation)
```suggestion
      state_was_set tr prin sess_id content
```

Reply:
Why not, but it is not [the style advocated by CONTRIBUTING.md](https://github.com/REPROSEC/dolev-yao-star-extrinsic/blob/main/CONTRIBUTING.md#inside-a-match), and it is not consistent with the rest of the codebase (which consistently uses indentations of two spaces without consideration for alignment). The rationale being that we want to avoid cases such as
```fstar
match x with
| SomeReallyLongAndExplicitNameThatOccupiesALotOfSpace foo ->
                                                       this will easily go too far on the right
| Bar bar ->
      this is not indented as the code two lines above
```

For this reason I will always prefer the two-space indentation style, but it's not worthy of having a debate on that, so do as you prefer. But if you want to keep the indent-to-align style, please choose between
```fstar
match x with
| Some y ->
      foo
```
and
```fstar
match x with
| Some y ->
       foo
```
as the two styles exist in this PR.